File: main_backend/src/services/email.service.js
const { emailServiceUrl, systemToken } = require('../config/env');
const axios = require('axios');

const sendEmail = async (message, subject, to) => {
    try {
        console.log("sendo email: ", message, subject, to);
        const response = await axios.post(`${emailServiceUrl}/send-email`,
            {
                message: message,
                subject: subject,
                to: to
            },
            {
                headers: {
                    Authorization: systemToken
                }
            }
        );

        console.log(response?.status, response?.data);

        if (response.status !== 200) {
            throw new Error(response.data.error);
        }
    } catch (error) {
        console.error(error.status, error.response.data);
        throw new Error(error);
    }
}

module.exports = { sendEmail };

File: frontend/dart/lib/services/email/emailService.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

/// Envia um e-mail via POST /send-email.
///
/// [token] é o JWT usado para autorização.
/// [to] é o e-mail do destinatário.
/// [subject] é o assunto do e-mail.
/// [message] é o corpo da mensagem.
Future<dynamic> sendEmail({
  required String token,
  required String to,
  required String subject,
  required String message,
}) async {
  final uri = Uri.parse('http://localhost:4002/send-email');

  try {
    final response = await http.post(
      uri,
      headers: {
        'Content-Type': 'application/json',
        'Authorization': token,
      },
      body: jsonEncode({
        'to': to,
        'subject': subject,
        'message': message,
      }),
    );

    if (response.statusCode == 200) {
      return jsonDecode(response.body.toString());
    } else {
      print(response.body);
      return {
        'error': true,
        'statusCode': response.statusCode,
        'body': response.body,
      };
    }
  } catch (e) {
    print(e.toString());
    return {
      'error': true,
      'exception': e.toString(),
    };
  }
}
